src/Rule/ImageHandler.php

```diff
@@ -0,0 +1,112 @@
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\RuleHandlerInterface;
+use Yiisoft\Validator\ValidationContext;
+
+final class ImageHandler implements RuleHandlerInterface
+{
+    public function validate(mixed $value, object $rule, ValidationContext $context): Result
+    {
+        if (!$rule instanceof Image) {
+            throw new UnexpectedRuleException(Image::class, $rule);
+        }
+
+        $result = new Result();
+
+        $info = $this->getImageInfo($value);
+        if (empty($info)) {
+            $result->addError($rule->getNotImageMessage(), ['attribute' => $context->getTranslatedAttribute()]);
+            return $result;
+        }
+
+        [$width, $height] = $info;
+
+        if ($rule->getWidth() !== null && $width !== $rule->getWidth()) {
+            $result->addError($rule->getNotExactlyWidthMessage(), [
+                'attribute' => $context->getTranslatedAttribute(),
+                'exactly' => $rule->getWidth(),
+            ]);
+        }
+        if ($rule->getHeight() !== null && $height !== $rule->getHeight()) {
+            $result->addError($rule->getNotExactlyHeightMessage(), [
+                'attribute' => $context->getTranslatedAttribute(),
+                'exactly' => $rule->getHeight(),
+            ]);
+        }
+        if ($rule->getMinWidth() !== null && $width < $rule->getMinWidth()) {
+            $result->addError($rule->getTooSmallWidthMessage(), [
+                'attribute' => $context->getTranslatedAttribute(),
+                'limit' => $rule->getMinWidth(),
+            ]);
+        }
+        if ($rule->getMinHeight() !== null && $height < $rule->getMinHeight()) {
+            $result->addError($rule->getTooSmallHeightMessage(), [
+                'attribute' => $context->getTranslatedAttribute(),
+                'limit' => $rule->getMinHeight(),
+            ]);
+        }
+        if ($rule->getMaxWidth() !== null && $width > $rule->getMaxWidth()) {
+            $result->addError($rule->getTooLargeWidthMessage(), [
+                'attribute' => $context->getTranslatedAttribute(),
+                'limit' => $rule->getMaxWidth(),
+            ]);
+        }
+        if ($rule->getMaxHeight() !== null && $height > $rule->getMaxHeight()) {
+            $result->addError($rule->getTooLargeHeightMessage(), [
+                'attribute' => $context->getTranslatedAttribute(),
+                'limit' => $rule->getMaxHeight(),
+            ]);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @psalm-return array{0:int,1:int}&array
+     */
+    private function getImageInfo(mixed $value): ?array
+    {
+        $filePath = $this->getFilePath($value);
+        if (empty($filePath)) {
+            return null;
+        }
+
+        if (!$this->isImageFile($filePath)) {
+            return null;
+        }
+
+        /**
+         * @psalm-var (array{0:int,1:int}&array)|null $info Need for PHP 8.0 only
+         */
+        $info = getimagesize($filePath);
```

Comment:
https://github.com/symfony/symfony/issues/46964

Could you add a comment about HEIC codec?